quickstart.py

```diff
@@ -309,6 +340,196 @@ def sendGetRequest(port):
   except urllib2.URLError:
     return None
 
+def stopDocker(config, forceStop = False):
+  """ Returns |False| on error. If Docker is not configured for restart and
+  |forceStop == False|, it will not be stopped and |True| (success) will be
+  returned
+  """
+  if not dockerRunning(config):
+    print "Docker already stopped."
+    return True
+
+  if not config["Docker"]["restart"] and not forceStop:
+    print "Not stopping Docker because it is not configured for restart"
+    return True
+
+  dockerApi = getDockerApi(config)
+  try:
+    dockerApi.stop(container = DOCKER_CONTAINER_NAME)
+  except:
+    print "Failed to stop Docker"
+    return False
+
+  if dockerRunning(config):
+    print "Failed to stop Docker"
+    return False
+
+  print "Docker stopped"
+  return True
+
+def startDocker(config, foreground = False, forceRebuild = False):
+  dockerApi = getDockerApi(config)
+  oldDockerConfig = getOldDockerConfigString()
+  imageInfo = dockerImageInfo(config)
+  rebuild = forceRebuild or not imageInfo or oldDockerConfig == None
+  if rebuild:
+    print "Building Docker image"
+    try:
+      dockerApi.remove_image(image = DOCKER_IMAGE_NAME, force = True)
+    except:
+      # Maybe the image just doesn't exist yet
+      pass
```

Comment:
This try/except form has a name called "the diaper pattern":

http://mike.pirnat.com/2009/05/09/the-diaper-pattern-stinks/

I mention it because it's kind of a funny anti-pattern name.